Realm/RLMQueryUtil.mm

```diff
@@ -141,41 +141,88 @@ void set_base_table(const Table*) override {}
 // A reference to a column within a query. Can be resolved to a Columns<T> for use in query expressions.
 class ColumnReference {
 public:
-    ColumnReference(RLMProperty* property, const std::vector<size_t>& links) :
-        m_property(property), m_links(links)
+    ColumnReference(RLMSchema *schema, RLMProperty* property, const std::vector<RLMProperty*>& links) :
+        m_links(links), m_property(property), m_schema(schema)
+    {
+    }
+    ColumnReference(RLMSchema *schema, RLMProperty* property) : m_property(property), m_schema(schema)
     {
     }
-    explicit ColumnReference(RLMProperty* property) : m_property(property) { }
 
-    template <typename T>
+    template <typename T, typename std::enable_if<!std::is_same<T, Link>::value>::type* = nullptr>
     auto resolve(Query& query) const
     {
         return table_for_query(query)->template column<T>(index());
     }
 
+    template <typename T, typename std::enable_if<std::is_same<T, Link>::value>::type* = nullptr>
```

Comment:
Since it doesn't need to be partially specialized, I think this could just be an explicit specialization for `<Link>`.